### PR TITLE
Mono Support for DisassemblyDiagnoser #541

### PR DESCRIPTION
**Pull request by morgan-kn**

### CODE REVIEW

**Comment by AndreyAkinshin** [comment]
Thanks, @morgan-kn!

File: src/BenchmarkDotNet.Core/Diagnosers/CopiedDataContracts.cs [hunk 1]
```diff
@@ -53,6 +53,8 @@ public class DisassembledMethod
         public string Problem { get; set; }
 
         public Map[] Maps { get; set; }
+
+        public string CommandLine { get; set; }
```
Comment (adamsitnik) [inline src/BenchmarkDotNet.Core/Diagnosers/CopiedDataContracts.cs]: the file is called `CopiedDataContracts` and it starts with `// keep it in sync with src\BenchmarkDotNet.Disassembler.x64\DataContracts.cs!`

could you please add the same field to the `src\BenchmarkDotNet.Disassembler.x64\DataContracts.cs` file?

File: src/BenchmarkDotNet.Core/Running/BenchmarkRunnerCore.cs [hunk 2]
```diff
@@ -366,7 +366,7 @@ private static BenchmarkReport RunCore(Benchmark benchmark, ILogger logger, Read
             {
                 logger.WriteLineInfo("// Run, Diagnostic [SeparateLogic]");
 
-                separateLogicCompositeDiagnoser.Handle(HostSignal.AfterAll, new DiagnoserActionParameters(null, benchmark, config));
+                separateLogicCompositeDiagnoser.Handle(HostSignal.SeparateLogic, new DiagnoserActionParameters(null, benchmark, config));
```
Comment (adamsitnik) [inline src/BenchmarkDotNet.Core/Running/BenchmarkRunnerCore.cs]: @morgan-kn very good catch! you have fixed my bug 💯

**Comment by AndreyAkinshin** [comment]
> Could we also give our users a warning if this tool is not present?

Absolutely! In fact, it's almost ready, I just want to check the detection logic on a few additional machines. It will be a part of the v0.10.13 release.

**Comment by adamsitnik** [comment]
@AndreyAkinshin awesome!

File: src/BenchmarkDotNet.Core/Diagnosers/CopiedDataContracts.cs (hunk 1 above)
Comment (morgan-kn) [reply src/BenchmarkDotNet.Core/Diagnosers/CopiedDataContracts.cs]: @adamsitnik should I fill this field? If so, I can suggest something like CommandLine = Environment.CommandLine in both DisassembleMethod and CreateEmpty.

File: src/BenchmarkDotNet.Core/Running/BenchmarkRunnerCore.cs (hunk 2 above)
Comment (morgan-kn) [reply src/BenchmarkDotNet.Core/Running/BenchmarkRunnerCore.cs]: @adamsitnik thanks =)

File: src/BenchmarkDotNet.Core/Diagnosers/CopiedDataContracts.cs (hunk 1 above)
Comment (adamsitnik) [reply src/BenchmarkDotNet.Core/Diagnosers/CopiedDataContracts.cs]: @morgan-kn no, just add it to the class. 

Some background: Windows disassembler must not have any dependencies to BDN, but it serializes the result to file. BDN has to be able to deserialize it, and also have no dependencies to the disassembler. Hence these two projects must duplicate same data contract, and we need this lame, manual files sync ;/

Comment (morgan-kn) [reply src/BenchmarkDotNet.Core/Diagnosers/CopiedDataContracts.cs]: @adamsitnik thank you =)

Comment (AndreyAkinshin) [reply src/BenchmarkDotNet.Core/Diagnosers/CopiedDataContracts.cs]: @adamsitnik, why do we have to sync these files manually?
In the `BenchmarkDotNet.Disassembler.x86` we use linked files:
```xml
<Compile Include="..\BenchmarkDotNet.Disassembler.x64\DataContracts.cs"
         Link="DataContracts.cs" />
```
What do you think about using the same trick for `BenchmarkDotNet.Core`?

Comment (adamsitnik) [reply src/BenchmarkDotNet.Core/Diagnosers/CopiedDataContracts.cs]: @AndreyAkinshin I remember that I have tried this approach, but for some reason, it did not work. I am not sure what was the problem that I hit back then ;/ It's most probably worth to give it a try once again

**Pull request merged**